### DISCUSSION TITLE
Is it possible to use ForEachShard and pipeline together ?

### DISCUSSION THREAD

**Discussion by bibicu123** [redis/go-redis#2611] [Q&A]
In case of ClusterClient is it possible to use pipelines and ForEachShard together?
I am having trouble figuring out how.

To give you more context:

we have a pipeline where we call: 
keys aaa*
keys bbb*
keys ccc*

So we should be able to call the pipeline ForEachShard in order to get all keys

**Comment by elena-kolevska** [comment]
Pipelines are always bound to a single shard because they exist on top of different connections, so it's not possible to reuse the same pipeline for multiple shards (to make this more obvious... imagine the shards on different physical servers). Also, if you're using this in production, I would advise against using `KEYS` since it is a blocking command. Instead, you should use [SCAN](https://redis.io/commands/scan/).
